ProfileService.FunctionalTests/Scenaries/Discount/ScenarioUpdateDiscountTests.cs

```diff
@@ -75,7 +119,7 @@ public async Task ScenarioUpdateDiscount()
             result
                 .Should()
                 .NotBeNull()
-                .Equals(discount);
+                .Equals(discount2);
```

Comment:
Are you sure that you compare values, not a references?

Normally (when not dealing with strings, that is), Equals compares values, while == compares object references.